ReSharper.FSharp/src/FSharp.Psi.Features/src/QuickFixes/ConvertTupleToArrayOrListElementsFix.fs

```diff
@@ -0,0 +1,43 @@
+﻿namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+
+open System.Text.RegularExpressions
+open JetBrains.ReSharper.Plugins.FSharp.Psi
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Parsing
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Psi.ExtensionsAPI
+open JetBrains.ReSharper.Psi.ExtensionsAPI.Tree
+open JetBrains.ReSharper.Psi.Util
+open JetBrains.ReSharper.Resources.Shell
+
+type ConvertTupleToArrayOrListElementsFix(expr: IFSharpExpression, actualType: string) =
+    inherit FSharpQuickFixBase()
+
+    let regex = Regex("'\w\d*( \* '\w\d*)+")
+    let arrayOrListExpr = ArrayOrListExprNavigator.GetByExpression(expr.IgnoreParentParens())
+    let isArray = arrayOrListExpr :? IArrayExpr
+
+    new (warning: AddTypeEquationError) =
+        ConvertTupleToArrayOrListElementsFix(warning.Expr, warning.ActualType)
+
+    override x.Text =
+        if isArray then "Convert tuple to array elements" else "Convert tuple to list elements"
+
+    override x.IsAvailable _ =
+        isValid expr && isNotNull arrayOrListExpr && regex.Match(actualType).Value = actualType
+
+    override x.ExecutePsiTransaction _ =
+        use writeCookie = WriteLockCookie.Create(expr.IsPhysical())
+        use disableFormatter = new DisableCodeFormatter()
+
+        match expr with
+        | :? ITupleExpr as tuple ->
+            for comma in tuple.Commas do
+                replace comma (FSharpTokenType.SEMICOLON.CreateLeafElement())
+
+            ModificationUtil.ReplaceChildRange(
```

Comment:
We expect the subexpressions to be in a `sequenceExpr`, so we're breaking the tree with replacing the tuple with its contents.
https://github.com/JetBrains/fsharp-support/blob/net211/ReSharper.FSharp/test/data/parsing/Expr%20-%20List%20-%20Seq%2001.fs.gold#L12

Btw, is it safe to replace a parent node with its child without copying here?

Reply:
I manually verified that after applying the fix, the tree structure remains correct. You can also explain the correctness by the fact that the sequence elements and ';' separators are recalculated from children on each call. 
![image](https://user-images.githubusercontent.com/26364714/105516760-4e8d0e80-5ce7-11eb-8603-18b4db18f1fa.png)

Reply:
> I manually verified that after applying the fix, the tree structure remains correct. 

Please show dumps of the trees for `[1;2]` and for before/after the quick fix on `[1,2]`. The tree after the quick fix is expected to be the same as `[1;2]`.

Reply:
Oh, yes, indeed, there is no ``sequenceExpr`` at all, sorry. I will fix it.